tests/test_operation.py

```diff
@@ -1370,6 +1370,21 @@ def test_generator_undefined(self):
         with pytest.raises(qml.operation.GeneratorUndefinedError):
             gate.generator()
 
+    def test_pow_zero(self):
+        """Test that the default of an operation raised to a zero power is an empty array."""
+        assert len(gate.pow(0)) == 0
```

Comment:
Would it make more sense for this to be the Identity operator? I could imagine that someone writes a function for an approximate matrix exponential: 

```
def matrix_exp_approx(op, N):
    for n in range(N): 
        coeff = 1/math.factorial(n)
        op_raise_n = op.pow(n) 
        res = qml.sum(res, (coeff*op_raise_n)) 
```

But this would not work with its current behaviour.

Reply:
This method is not meant to be used independently but used in the decomposition of a `Pow` operator wrapper that will be created soon. 

Since it is essentially a `decomposition`, I'm mirroring the same behaviour of `decomposition`.

```
def matrix_exp_approx(op, N):
    for n in range(N): 
        coeff = 1/math.factorial(n)
        op_raise_n = op ** n
        # option 2
        op_raise_n = pow(op, n)
        res = qml.sum(res, (coeff*op_raise_n)) 
```

Reply:
No identity, just an empty list. The identity decomposes to an empty list anyway.